src/app/components/shared/menu/link/link.component.ts

```diff
@@ -93,6 +107,21 @@ export class MenuLinkComponent implements OnChanges {
     if (!this.isInternalLink) {
       this.handleExternalLink();
     }
+
+    /*
+     * Components outside of router-outlet are unable to read the query/route
+     * params of the page component. So we use this bypass, check the service
+     * for more details
+     */
+    this.sharedRoute.activatedRoute$
+      .pipe(
+        map((route): ActivatedRouteSnapshot => route.snapshot),
```

Comment:
i still think this can be packaged into `SharedActivatedRouteService`.

plus a name would make the intent a little more discernable... `this.sharedRoute.getSnapshot$`